x-pack/platform/plugins/shared/triggers_actions_ui/public/application/components/simple_connector_form.tsx

```diff
@@ -54,12 +56,14 @@ const getFieldConfig = ({
   isUrlField = false,
   defaultValue,
   type,
+  allowDomainWithoutDots = false,
 }: {
   label: string;
   isRequired?: boolean;
   isUrlField?: boolean;
   defaultValue?: string | string[];
   type?: keyof typeof FIELD_TYPES;
+  allowDomainWithoutDots: boolean;
```

Comment:
I don't think we should handle this at the component props level. Have you considered extending `ConfigFieldSchema` with an additional property, perhaps like `validator`, to allow for custom validation rules for any field?